Core/Core/SwiftUIViews/OptionalDatePicker.swift

```diff
@@ -19,47 +19,111 @@
 import SwiftUI
 
 public struct OptionalDatePicker<Label: View>: View {
+    let initial: Date
     let label: Label
     let min: Date
     let max: Date
-    let initial: Date
+    let placeholder: Text
+
     @Binding var selection: Date?
+
     @State var showDatePicker = false
 
-    public init(selection: Binding<Date?>, min: Date? = nil, max: Date? = nil, initial: Date, @ViewBuilder label: () -> Label) {
+    public init(
+        placeholder: Text = Text("--", bundle: .core),
+        selection: Binding<Date?>,
+        min: Date? = nil,
+        max: Date? = nil,
+        initial: Date,
+        @ViewBuilder label: () -> Label
+    ) {
         self.label = label()
         self.min = min ?? .distantPast
         self.max = max ?? .distantFuture
-        self.initial = initial
+        self.initial = Swift.min(self.max, Swift.max(self.min, initial))
+        self.placeholder = placeholder
         self._selection = selection
     }
 
-    public var body: some View { VStack(spacing: 0) {
-        ButtonRow(action: {
-            if self.showDatePicker == false, self.selection == nil {
-                self.selection = self.initial
-            }
-            self.showDatePicker.toggle()
-        }, content: {
-            label
-            Spacer()
-            Text(selection?.dateTimeString ?? NSLocalizedString("--", bundle: .core, comment: ""))
-                .font(.regular14)
-            Spacer().frame(width: 12)
-            if selection != nil {
-                Button(action: { self.selection = nil }, label: {
+    public var body: some View {
+        if #available(iOSApplicationExtension 14, *) {
+            compact
```

Comment:
The compact datepicker shows inline in the row instead of having the text date in a button that toggles a wheel picker like before.